tests/test_FiberFlat.py

```diff
@@ -0,0 +1,67 @@
+import sys
+import unittest
+import pickle
+
+import numpy as np
+
+import lsst.utils.tests
+import lsst.pex.exceptions
+import lsst.geom
+import lsst.afw.image
+import lsst.afw.image.testUtils
+
+import pfs.drp.stella as drpStella
+import lsst.afw.image as afwImage
+from pfs.drp.stella.constructFiberFlatTask import ConstructFiberFlatTask
+
+
+class FiberFlatTestCase(lsst.utils.tests.TestCase):
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def testExtractNonOptimal(self):
+        """ Tests the ConstructFiberFlatTask.correctLowVarianceImage method
+        """
+        width = 4096
+        height = 4176
+
+        defaultVarValue = 10.0
+        defaultImageValue = 1.0
+        defaultMaskValue = 0x0
+        minVariance = 4.0
+
+        affectedPixelRow = 123
+        affectedPixelCol = 456
+
+        maskedImage = afwImage.MaskedImageF(width, height)  # Image for extraction
+        maskedImage.getImage().getArray()[:] = defaultImageValue
+        maskedImage.getMask().getArray()[:] = defaultMaskValue
+        maskedImage.getVariance().getArray()[:] = defaultVarValue
+        maskedImage.getVariance().getArray()[affectedPixelRow, affectedPixelCol] = defaultImageValue
+
+        cfft = ConstructFiberFlatTask()
+        cfft.correctLowVarianceImage(maskedImage, minVariance)
+
+        imgArr = maskedImage.getImage().getArray()
+        varArr = maskedImage.getVariance().getArray()
+        maskArr = maskedImage.getMask().getArray()
+        isLowVar = varArr < minVariance
+
+        self.assertFalse(np.any(varArr[isLowVar]))
```

Comment:
Not sure `varArr[]` belongs here (it doesn't matter if everything is good, but may blow up rather than fail if things are bad).

Reply:
Can you clarify further Paul? I'm not sure what needs to be fixed.

Reply:
Won't `np.any(varArr[isLowVar])` blow up if `isLowVar` contains any `True` elements, since `varArr` is a `float` array rather than `bool`?

Reply:
Agreed. Updated to ``self.assertFalse(np.any(isLowVar))``. Any better suggestions, please tell me.